File: backend/src/models/Trip.js
import mongoose from 'mongoose';

const tripSchema = new mongoose.Schema(
  {
    route_id: {
      type: String,
      required: true,
      trim: true
    },
    origin: {
      type: String,
      required: true,
      trim: true,
      default: 'Addis Ababa'
    },
    destination: {
      type: String,
      required: true,
      trim: true,
      default: 'Hawassa'
    },
    vehicle_id: {
      type: String,
      required: true,
      trim: true
    },
    driver_id: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true
    },
    departure_time: {
      type: Date,
      required: true
    },
    arrival_time: {
      type: Date,
      required: true
    },
    price_per_seat: {
      type: Number,
      required: true,
      min: 0
    },
    status: {
      type: String,
      enum: ['scheduled', 'in_transit', 'completed', 'cancelled'],
      default: 'scheduled'
    }
  },
  {
    timestamps: { createdAt: 'created_at', updatedAt: 'updated_at' }
  }
);

tripSchema.set('toJSON', {
  transform: (doc, ret) => {
    ret.id = ret._id;
    delete ret._id;
    delete ret.__v;
    return ret;
  }
});

const Trip = mongoose.model('Trip', tripSchema);
export default Trip;

File: driver_app/lib/models/trip.dart
class Trip {
  final String id;
  final String routeId;
  final String origin;
  final String destination;
  final String vehicleId;
  final String driverId;
  final DateTime departureTime;
  final DateTime arrivalTime;
  final double pricePerSeat;
  final String status;

  Trip({
    required this.id,
    required this.routeId,
    required this.origin,
    required this.destination,
    required this.vehicleId,
    required this.driverId,
    required this.departureTime,
    required this.arrivalTime,
    required this.pricePerSeat,
    required this.status,
  });

  factory Trip.fromJson(Map<String, dynamic> json) {
    return Trip(
      id: json['id']?.toString() ?? '',
      routeId: json['route_id']?.toString() ?? '',
      origin: json['origin']?.toString() ?? '',
      destination: json['destination']?.toString() ?? '',
      vehicleId: json['vehicle_id']?.toString() ?? '',
      driverId: json['driver_id']?.toString() ?? '',
      departureTime: DateTime.tryParse(json['departure_time']?.toString() ?? '') ??
          DateTime.now(),
      arrivalTime: DateTime.tryParse(json['arrival_time']?.toString() ?? '') ??
          DateTime.now(),
      pricePerSeat: (json['price_per_seat'] as num?)?.toDouble() ?? 0.0,
      
      status: json['status']?.toString() ?? 'scheduled',
    );
  }
}